### ISSUE TITLE
Allow this.state in setState callback

### ISSUE THREAD

**Issue by tobilen** [BE-Webdesign/eslint-plugin-react-functional-set-state#1]
Currently, this raises a linting error:
```js
this.setState(prevState => prevState, () => console.log(this.state))
```

Using this.state in the callback should be allowed

**Comment by BE-Webdesign** [comment]
Hi @tobilen ,

Thank you for the report. This linting plugin is not complete yet and the only rule currently is the no-this-state-props, which is to lint the use of using this.state or this.props inside of a call to this.setState(). So the linter is working as expected, but probably not as part of what you are trying to accomplish. There are no docs or anything set up yet to explain this as well so I apologize for the confusion but appreciate the report.

Do you have a reference to where providing two function to setState is applicable? I have not seen that before.

**Comment by tobilen** [comment]
hey @BE-Webdesign ,

thanks for the quick reply. no need to apologize for anything. its a great plugin.

As to your question, i cant really point you to anywhere in the docs, but the main reason not to reference `this.state` is because setState works async and gets compounded when called multiple times during a cycle. since the callback does not follow this pattern and only gets called when state has been settled, i dont see why we wouldnt want to use it. please correct me if im wrong.

p.s.: i opened a PR with the changes

**Comment by tobilen** [comment]
>Do you have a reference to where providing two function to setState is applicable?

second function is an optional parameter. its expects a function which gets called when setState is done modifying the state.
https://facebook.github.io/react/docs/react-component.html#setstate

**Comment by BE-Webdesign** [comment]
> https://facebook.github.io/react/docs/react-component.html#setstate

Cool thanks, looks good. I will check out the PR you made soonish.

**Comment by tobilen** [comment]
is this happening?

**Comment by BE-Webdesign** [comment]
Lol sorry. Yes it will happen. I plan on getting this lint plugin into good shape over the weekend. I will also be pursuing trying to get this added to the main eslint react plugin as well so any work you do here will probably be transferred into that as well. Sound good?

**Comment by BE-Webdesign** [comment]
Closed by #2. Thank you @tobilen.
